### PR TITLE
Added example to show RESTful JSON POST

### PR DESCRIPTION
**Pull request by slightfoot**
I keep seeing questions on the flutter gitter chat about how to do this. So I think it best to be shown in the docs.

### CODE REVIEW

**Comment by slightfoot** [comment]
I signed it!

File: networking.md [hunk 1]
```diff
@@ -79,6 +79,42 @@ Map, List, or List of Maps containing simple values, to the `encode` method:
...
+    HttpClient client, String url, Map<String, dynamic> jsonBody,
+    [String accessToken]) async {
+  final String requestBody = JSON.encode(jsonBody);
+  HttpClientRequest request = await client.postUrl(Uri.parse(url))
+    ..headers.add(HttpHeaders.ACCEPT, ContentType.JSON)
+    ..headers.contentType = ContentType.JSON
+    ..headers.contentLength = requestBody.length
+    ..headers.chunkedTransferEncoding = false;
+  if (accessToken != null) {
+    request.headers.add(HttpHeaders.AUTHORIZATION, 'Bearer $accessToken');
```
Comment (Hixie) [inline networking.md]: you can use `client.addCredentials` to do this more idiomatically.

**Comment by slightfoot** [comment]
@filiph Yes, I looked at @Hixie's suggestion, but there's currently no implementation of `Bearer` authorization only `Basic` and `Digest`, it is also not possible to extend the built in classes to add this functionality at present. The simple one liner above I think is fine for the example at present.

**Comment by filiph** [comment]
Sorry for the delay. Merging!

**Comment by goderbauer** [comment]
This broke the website build (wasn't caught by travis because of the delay between posting the PR and merging):


```
hint • 'UTF8' is deprecated and shouldn't be used at example/networking_md_1.dart:3:28 • deprecated_member_use
  hint • 'JSON' is deprecated and shouldn't be used at example/networking_md_1.dart:3:34 • deprecated_member_use
```

I'm sending out a PR to fix this in a minute.

**Pull request merged**